### DISCUSSION TITLE
Warn old patch players about score non-submission

### DISCUSSION THREAD

**Discussion by Non-Binary-Programmer** [ppy/osu#27006] [Ideas]
The current behavior of osu is that if a player completes a score while online, but on an old patch, everything looks normal on the client side but the score does not submit. I would suggest that a warning message be added if a user completes a score on an old patch, to remind them that they need to update.

**Comment by bdach** [comment]
how are we supposed to add a warning message to an old release that people have already downloaded and we cannot change without *them updating the client to a newer release which makes this entire idea circular*

we can only do this going forward, have talked about this internally

**Reply by Non-Binary-Programmer** [reply]
If you can't tell when you're on an old patch, _how does the auto updater work?_

To clarify, I'm only proposing doing this for players who are online, but forgot / chose not to apply the new patch. (this did happen to me, and it made me lose a personal top ten score)

**Reply by bdach** [reply]
the point i was trying to make is that we can only add such a warning to future releases, it is too late to do anything with past ones

the update delivery mechanism is not a vehicle that can deliver arbitrary messages to the users

**Reply by Non-Binary-Programmer** [reply]
Well then, I guess my suggestion is that a warning is added to the next public patch (and subsequent patches) for when that patch becomes obsoleted, so that players don't accidentally keep playing on a patch where their scores are not submitting. Sorry for the confusing wording.

**Comment by peppy** [comment]
We will be turning off lazer.ppy.sh soon which should give them a better hint that things are not working.

**Reply by Non-Binary-Programmer** [reply]
This happened for me when the old patch was after the migration away from lazer.ppy.sh
